cdap-explore/src/main/java/co/cask/cdap/explore/guice/ExploreRuntimeModule.java

```diff
@@ -231,8 +231,10 @@ protected void configure() {
         setupClasspath();
 
         // Set local tmp dir to an absolute location in the twill runnable otherwise Hive complains
+        String localScratchPath = System.getProperty("java.io.tmpdir") + File.separator +
```

Comment:
Read the code some more. This part is only for distributed mode. In standalone, we have the following:

```

        File hiveDataDir = new File(cConf.get(Constants.Explore.LOCAL_DATA_DIR));

        // The properties set using setProperty will be included to any new HiveConf object created,
        // at the condition that the configuration is known by Hive, and so is one of the HiveConf.ConfVars
        // variables.

        System.setProperty(HiveConf.ConfVars.SCRATCHDIR.toString(),
                           new File(hiveDataDir, cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsolutePath());
```

so it's just using the data dir.